### PR TITLE
add unit tests

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Adds Unit tests for ./server/app/auth.go. 
Adds Unit tests for ./server/auth/auth.go. 
These will be used as samples for other server tests.

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/421

### CODE REVIEW

**Comment by sbishel** [comment]
These will be used as samples for remaining unit test tickets.

**Review by harshilsharma63 — APPROVED** [review]

File: server/app/helper.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (jespino) [inline server/app/helper.go:1]: This helper file is compiling everything inside the app package when is not compiling the tests, so... we are importing here, in the production binary the mocks a some not-needed code.

File: server/app/auth_test.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	}{
+		{"fail, missing login information", "", "", "", "", true},
+		{"fail, invalid username", "badUsername", "", "", "", true},
+		{"fail, invalid email", "", "badEmail", "", "", true},
+		{"fail, invalid password", "testUsername", "", "badPassword", "", true},
+		{"success, using username", "testUsername", "", "testPassword", "", false},
+		{"success, using email", "", "testEmail", "testPassword", "", false},
+	}
+
+	th.Store.EXPECT().GetUserByUsername(gomock.Eq("badUsername")).Return(nil, errors.New("Bad Username"))
```
Comment (jespino) [inline server/app/auth_test.go:41]: I'm not sure (I'll check), but I think if you are expecting one specific parameter you can use just `...GetUserByUsername("badUsername")...`

**Review by jespino — APPROVED** [review]
Now LGTM.

**Pull request merged**